### PR TITLE
Make RuntimeManager implicit to MultiParentCasper

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

The title says it all

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3693

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by KentShikama** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/Casper.scala [hunk 1]
```diff
@@ -67,8 +67,7 @@ trait MultiParentCasper[F[_]] extends Casper[F, IndexedSeq[BlockHash]] {
   // this initial fault weight combined with our fault tolerance threshold t.
   def normalizedInitialFault(weights: Map[Validator, Long]): F[Float]
   def lastFinalizedBlock: F[BlockMessage]
-  // TODO: Refactor hashSetCasper to take a RuntimeManager[F] just like BlockStore[F]
-  def getRuntimeManager: F[Option[RuntimeManager[F]]]
+  def getRuntimeManager: F[RuntimeManager[F]]
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/Casper.scala:71]: Changing the API for getRuntimeManager should have been in a different commit but I hope that isn't too much extra to swallow.

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala [hunk 2]
```diff
@@ -110,8 +110,7 @@ object BlockAPI {
     ): Effect[F, ListeningNameDataResponse] =
       for {
         mainChain           <- getMainChainFromTip[F](depth)
-        maybeRuntimeManager <- casper.getRuntimeManager
-        runtimeManager      = maybeRuntimeManager.get // This is safe. Please reluctantly accept until runtimeManager is no longer exposed.
+        runtimeManager      <- casper.getRuntimeManager
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala:113]: Just announcing changes that stem from the getRuntimeManager API change

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala [hunk 3]
```diff
@@ -144,9 +143,8 @@ object BlockAPI {
         implicit casper: MultiParentCasper[F]
     ): Effect[F, ListeningNameContinuationResponse] =
       for {
-        mainChain           <- getMainChainFromTip[F](depth)
-        maybeRuntimeManager <- casper.getRuntimeManager
-        runtimeManager      = maybeRuntimeManager.get // This is safe. Please reluctantly accept until runtimeManager is no longer exposed.
+        mainChain      <- getMainChainFromTip[F](depth)
+        runtimeManager <- casper.getRuntimeManager
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala:147]: Just announcing changes that stem from the getRuntimeManager API change

**Comment by KentShikama** [comment]
bors try-

**Comment by KentShikama** [comment]
bors try

File: casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala [hunk 4]
```diff
@@ -171,86 +255,14 @@ class BlocksResponseAPITest
         _         <- casperRef.set(casperEffect)
         cliqueOracleEffect = SafetyOracle
           .cliqueOracle[Task](Sync[Task], logEff, metricsEff, spanEff)
-        blocksResponse <- BlockAPI.getBlocks[Task](Some(Int.MaxValue))(
+        blocksResponse <- BlockAPI.getBlocks[Task](Some(2))(
```
Comment (tymm) [inline casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala:258]: Should this be a separate commit?

File: casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala [hunk 5]
```diff
@@ -171,86 +255,14 @@ class BlocksResponseAPITest
...
                            Sync[Task],
                            casperRef,
                            logEff,
                            cliqueOracleEffect,
                            blockStore
                          )
-      } yield blocksResponse.right.get.length should be(8) // TODO: Switch to 4 when we implement block height correctly
-  }
-
-  it should "return until depth" in withStorage { implicit blockStore => implicit blockDagStorage =>
```
Comment (tymm) [inline casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala:184]: This test is gone? Should be a separate commit as well.

**Review by tymm — APPROVED** [review]

**Review by itegulov — APPROVED** [review]
LGTM

File: casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala (hunk 4 above)
Comment (KentShikama) [reply casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala:258]: Nope; it is just a confusing diff

File: casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala (hunk 5 above)
Comment (KentShikama) [reply casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala:184]: No test has been removed
